### PR TITLE
[lldb] Implement TypeInfoProvider::getId using the type system as the ID

### PR DESCRIPTION
**Pull request by augusto2112**
rdar://80001304

### CODE REVIEW

**Comment by augusto2112** [comment]
https://github.com/apple/swift/pull/64041

@swift-ci test

**Comment by augusto2112** [comment]
https://github.com/apple/swift/pull/64041

@swift-ci test

File: lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp [hunk 1]
```diff
@@ -588,7 +588,12 @@ class LLDBTypeInfoProvider : public swift::remote::TypeInfoProvider {
 public:
   LLDBTypeInfoProvider(SwiftLanguageRuntimeImpl &runtime,
                        TypeSystemSwift &typesystem)
-      : m_runtime(runtime), m_typesystem(typesystem) {}
+      : m_runtime(runtime),
+        // Always use the typeref type system so we have fewer cache
+        // invalidations.
+        m_typesystem(typesystem.GetTypeSystemSwiftTypeRef()) {}
+
+  uint64_t getId() override { return (uint64_t)&m_typesystem; }
```
Comment (adrian-prantl) [inline lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp]: See my comment in the other review: maybe use `void *`?

**Comment by augusto2112** [comment]
https://github.com/apple/swift/pull/64056

@swift-ci smoke test

**Comment by augusto2112** [comment]
@swift-ci test

**Comment by augusto2112** [comment]
@swift-ci test

**Comment by augusto2112** [comment]
@swift-ci test

**Pull request merged**